### PR TITLE
Adds our traits define file to the linter steps

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Moves all of the Skyrat traits to one central declarations file and adds a linter step to make sure defined traits are also in the global trait list.

## Changelog

Nothing player facing

### CODE REVIEW

File: code/__DEFINES/~skyrat_defines/traits/declarations.dm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+/// Cyborgs with unique sprites for when they get totally broken down.
+#define TRAIT_R_UNIQUEWRECK	"unique_wreck"
+/// Or when tipped over.
+#define TRAIT_R_UNIQUETIP	"unique_tip"
+/// 64x32 skins
+#define TRAIT_R_WIDE		"wide_borg"
+/// 32x64 skins
+#define TRAIT_R_TALL		"tall_borg"
+/// Any model small enough to reject the shrinker upgrade.
+#define TRAIT_R_SMALL		"small_chassis"
```
Comment (GoldenAlpharex) [inline code/__DEFINES/~skyrat_defines/traits/declarations.dm]:
```suggestion
#define TRAIT_R_UNIQUEWRECK "unique_wreck"
/// Or when tipped over.
#define TRAIT_R_UNIQUETIP	"unique_tip"
/// 64x32 skins
#define TRAIT_R_WIDE "wide_borg"
/// 32x64 skins
#define TRAIT_R_TALL "tall_borg"
/// Any model small enough to reject the shrinker upgrade.
#define TRAIT_R_SMALL "small_chassis"
```

File: code/__DEFINES/~skyrat_defines/traits/declarations.dm [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+Remember to update _globalvars/traits.dm if you're adding/removing/renaming traits.
+*/
+
+//mob traits
+
+// Defines for some extra traits
+#define TRAIT_NO_HUSK "no_husk"
+#define TRAIT_NORUNNING "norunning"		// You walk!
+#define TRAIT_EXCITABLE	"wagwag" //Will wag when patted!
+#define TRAIT_OXYIMMUNE	"oxyimmune"		// Immune to oxygen damage, ideally give this to all non-breathing species or bad stuff will happen
```
Comment (GoldenAlpharex) [inline code/__DEFINES/~skyrat_defines/traits/declarations.dm]:
```suggestion
#define TRAIT_NO_HUSK "no_husk"
#define TRAIT_NORUNNING "norunning" // You walk!
#define TRAIT_EXCITABLE "wagwag" //Will wag when patted!
#define TRAIT_OXYIMMUNE	"oxyimmune" // Immune to oxygen damage, ideally give this to all non-breathing species or bad stuff will happen
```

**Review by GoldenAlpharex — DISMISSED** [review]

File: code/__DEFINES/~skyrat_defines/traits/declarations.dm [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+// Defines for some extra inherent traits
+#define TRAIT_REVIVES_BY_HEALING "trait_revives_by_healing"
+#define TRAIT_ROBOTIC_DNA_ORGANS "trait_robotic_dna_organs"
+
+//Defines for model features, set in the model_features list of a robot model datum. Are they a dogborg? Is the model small? etc.
+/// Cyborgs with unique sprites for when they get totally broken down.
+#define TRAIT_R_UNIQUEWRECK "unique_wreck"
+/// Or when tipped over.
+#define TRAIT_R_UNIQUETIP	"unique_tip"
```
Comment (vinylspiders) [inline code/__DEFINES/~skyrat_defines/traits/declarations.dm]:
```suggestion
#define TRAIT_R_UNIQUETIP "unique_tip"
```

**Review by vinylspiders — COMMENTED** [review]
I don't know what is going on with these

**Review by jjpark-kb — APPROVED** [review]

**Pull request merged**